components/gameboard/DiscardPile.tsx

```diff
@@ -1,13 +1,14 @@
 import styles from "./Piles.module.css";
-import { PileProps } from "./types";
+import { DiscardPileProps } from "./types";
 
-function DiscardPile({ onClick }: PileProps) {
+function DiscardPile({ onClick, card }: DiscardPileProps) {
   return (
     <>
       <section className={styles.container}>
         <h2 className={styles.discardHeadline}>Discard Pile</h2>
         <img
-          src="/cards/blank.png"
+          key={card ? card.value : ""}
+          src={card ? card.imgSrc : "/cards/blank.png"}
```

Comment:
The `DiscardPileProps` says, that `card` is not optional (`card: Card;`). Thus, it shouldn't be required to check if `card` exists.
You set `card={null}` in your story, instead, I would make it optional and remove `card={null}`:
`card?: Card;`.

Reply:
It's required here, because the initial state when the game hasn't started is, that there's no card in the discard pile backend.